File: code/src/main/java/workout_manager/model/Days.java
package workout_manager.model;

import java.text.SimpleDateFormat;
import java.time.LocalDate;
import java.util.Calendar;
import java.util.Date;

/**
 * Creates a enum for the days of the week
 * 
 * @version Spring 2022
 * @author wtjracer
 */
public enum Days {
    MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY,
    SATURDAY, SUNDAY;

    public static Date getDate(Date day, String daySelected){
        int dayToFind = Days.stringToEnum(daySelected);
        System.out.println(daySelected);
        System.out.println(dayToFind);
        Calendar calendar = Calendar.getInstance();
        int difference =  dayToFind - calendar.get(Calendar.DAY_OF_WEEK);
        System.out.println(calendar.get(Calendar.DAY_OF_WEEK));
        System.out.println(difference);
        calendar.setTime(day); 
        calendar.add(Calendar.DATE, difference);
        System.out.print(calendar.getTime());
        return calendar.getTime();
    }
    private static int stringToEnum(String day){
        if (day.toUpperCase().equals(Days.SUNDAY.toString())) {
            return 1;
        }
        if (day.toUpperCase().equals(Days.MONDAY.toString())) {
            return 2;
        }
        if (day.toUpperCase().equals(Days.TUESDAY.toString())) {
            return 3;
        }
        if (day.toUpperCase().equals(Days.WEDNESDAY.toString())) {
            return 4;
        }
        if (day.toUpperCase().equals(Days.THURSDAY.toString())) {
            return 5;
        }
        if (day.toUpperCase().equals(Days.FRIDAY.toString())) {
            return 6;
        }
        return 7;
    }
}

File: code/Server/Days.py
#!/usr/bin/env python3
from enum import Enum

#*
# * Creates a enum for the days of the week
# * 
# * @version Python Spring 2022
# * @author Jordan Wagner
# 
class Days(Enum):
    MONDAY = 0
    TUESDAY = 1
    WEDNESDAY = 2
    THURSDAY = 3
    FRIDAY = 4
    SATURDAY = 5
    SUNDAY = 6

    #*
    # * Creates a enum for the days of the week froma  string
    # * @param day the day to get the enum for
    # * @returns the appropriate enum
    # 
    def fromString(day):
        if day in 'MONDAY':
            return Days.MONDAY
        elif day in ('TUESDAY'):
            return Days.TUESDAY
        elif day in ('WEDNESDAY'):
            return Days.WEDNESDAY
        elif day in ('THURSDAY'):
            return Days.THURSDAY
        elif day in ('FRIDAY'):
            return Days.FRIDAY
        elif day in ('SATURDAY'):
            return Days.SATURDAY
        elif day in ('SUNDAY'):
            return Days.SUNDAY
        else:
            raise NotImplementedError

        #*
    # * Creates a string for the days of the week from an enum
    # * @param day the day to get the string for
    # * @returns the appropriate string
    # 
    def toString(day):
        if day == Days.MONDAY:
            return 'MONDAY'
        elif day ==  Days.TUESDAY:
            return 'TUESDAY'
        elif day == Days.WEDNESDAY:
            return 'WEDNESDAY'
        elif day == Days.THURSDAY:
            return 'THURSDAY'
        elif day == Days.FRIDAY:
            return 'FRIDAY'
        elif day == Days.SATURDAY:
            return 'SATURDAY'
        elif day == Days.SUNDAY:
            return 'SUNDAY'
        else:
            raise NotImplementedError
